anise/src/almanac/mod.rs

```diff
@@ -164,9 +164,27 @@ impl Almanac {
         }
 
         if let Ok(metadata) = Metadata::decode_header(&bytes) {
+            // Use `try_from` to validate the dataset type
+            let dataset_type =
+                DataSetType::try_from(metadata.dataset_type as u8).map_err(|err| {
+                    AlmanacError::GenericError {
+                        err: format!("Invalid dataset type: {}", err),
```

Comment:
```suggestion
                        err: format!("Invalid dataset type: {err}"),
```

I'm surprised that clippy didn't complain about using the long form version of formatting simple variables, but regardless, I try to use that form when possible.

Reply:
Suggestion incorporated in https://github.com/nyx-space/anise/pull/418/commits/601854543968ee1a27a268491aef70359baab227.